### ISSUE TITLE
the C_obs obtained after running the following code are all 0

### ISSUE THREAD

**Issue by zhangyu2234** [InhwanBae/EigenTrajectory#8]
Dear author, first of all, thank you for providing such a wonderful work, but I have a question. When processing the ETH dataset, the C_obs obtained after running the following code are all 0. Can you tell me why this is? Is there an error in one of my steps?
        
        
        n_ped = obs_traj.size(0)

        # Filter out static trajectory
        mask = (obs_traj[:, -1] - obs_traj[:, -3]).div(2).norm(p=2, dim=-1) > self.static_dist
        obs_m_traj = obs_traj[mask]
        obs_s_traj = obs_traj[~mask]
        pred_m_traj_gt = pred_traj[mask] if pred_traj is not None else None
        pred_s_traj_gt = pred_traj[~mask] if pred_traj is not None else None

        # Projection
        C_m_obs, C_m_pred_gt = self.ET_m_descriptor.projection(obs_m_traj, pred_m_traj_gt)
        C_s_obs, C_s_pred_gt = self.ET_s_descriptor.projection(obs_s_traj, pred_s_traj_gt)
        C_obs = torch.zeros((self.k, n_ped), dtype=torch.float, device=obs_traj.device)
        C_obs[:, mask], C_obs[:, ~mask] = C_m_obs, C_s_obs  # KN

**Comment by InhwanBae** [comment]
Hi @zhangyu2234,

Thank you for your interest in my work! I’ve just re-run the code to double-check, and I can confirm that the values are being processed correctly. Could you let me know which baseline model you’re using? In the code, a temporary zero-filled `C_obs` is created and then filled using `C_m_obs` and `C_s_obs`. Could you verify if `C_m_obs` and `C_s_obs` are also zero? If you could provide more details about the issue, I’d be happy to help with debugging.

**Comment by InhwanBae** [comment]
Hi @zhangyu2234 ,

Thanks for your interest in my work! I've looked into this issue again, and I can confirm that `C_obs` is being correctly filled. However, I did find cases where `C_obs` becomes zero, specifically, when the pedestrian is standing still, especially in the ETH scene. In this case, all values are filled with zeros. While this may seem unusual at first, it's actually expected behavior because the xy coordinates remain constant and do not change over time.

I'm closing this issue for now. Feel free to open a new one if you have any further questions!
